### ISSUE TITLE
Automatically merge exact duplicates

### ISSUE THREAD

**Issue by joeroe** [xronos-ch/xronos.rails#309]
Exactly duplicated records should be automatically merged after save.

This should be sufficient to eliminate them all, but just to be safe, and to deal with the large backlog of duplicates we have now, we could also consider a task that periodically or on a trigger runs the same check on all records.

**Comment by joeroe** [comment]
Some interesting thoughts from the robot:

> 1. "Implement auto-merge of exact duplicates on save" — Will require a Site::Merger (or generic Duplicable::Merger) class that:
> - Takes a Relation of exact duplicates.
> - Picks the canonical (oldest non-superseded, per #309).
> - Supersedes the others (set superseded_by, revision_comment, paper_trail_event = "merge").
> - Reassigns children via reassign_associations.
> - Wired into an after_save callback or a one-off task.
> - No DuplicatesController for now — that path is uncertain.

**Comment by MartinHinz** [comment]
Looking at #496, I think the supersession/event model is a very good foundation for this. It gives us canonical redirects, an explicit supersession state, an event log, and transactional reassignment of child associations.

For automatic exact-duplicate merging, I would still add one cautious layer on top. Before calling supersede!, the merger should inspect the relevant child associations and decide whether each child is already represented on the canonical record, can be safely reassigned, or creates a conflict. The first two cases could be handled automatically; conflicts should stop the automatic merge and be flagged for manual review.

So I would see #496 as the low-level supersession mechanism, and #309 as a later merger/backfill layer that uses it conservatively.

**Comment by joeroe** [comment]
I agree. I'm also thinking now thinking that the logic added to automatically detect and reassign associations in #496 (`reassign_associations`) is too risky. We should explicitly specify which associations needed to be updated and how in a callback (e.g. `after_supersession`) on individual models.
